lib/Backend/DecompileFunction.cpp

```diff
@@ -794,26 +761,27 @@ CCodeGenerator::getCustomOpcodeToken(const llvm::CallInst *Call) const {
   if (isCallToTagged(Call, FunctionTags::ModelCast)) {
     // First argument is a string containing the base type
     auto *CurArg = Call->arg_begin();
-    QualifiedType CurType = deserializeFromLLVMString(CurArg->get(), Model);
+    model::UpcastableType CurType = deserializeFromLLVMString(CurArg->get(),
+                                                              Model);
 
     // Second argument is the base llvm::Value
     ++CurArg;
     llvm::Value *BaseValue = CurArg->get();
 
     // Emit the parenthesized cast expr, and we are done
     std::string StringToCast = rc_recur getToken(BaseValue);
-    rc_return buildCastExpr(StringToCast, TypeMap.at(BaseValue), CurType);
+    rc_return buildCastExpr(StringToCast, *TypeMap.at(BaseValue), *CurType);
   }
 
   if (isCallToTagged(Call, FunctionTags::AddressOf)) {
     // First operand is the type of the value being addressed (should not
     // introduce casts)
-    QualifiedType ArgType = deserializeFromLLVMString(Call->getArgOperand(0),
-                                                      Model);
+    const model::UpcastableType
```

Comment:
1. Why `const`?
2. This `.cpp` file has `using namespace model`, I think, can we drop `model::`?

Reply:
1. Because the world is a sad place. The binary here is const, so this is the only way to use it with `skipTypedefs`. Until we look into the delayed constness rework that is.
2. Sad but ok.